### DISCUSSION TITLE
Make the Empty Items view clickable

### DISCUSSION THREAD

**Discussion by NarendraKuruva** [adobe/react-spectrum#6995] [Q&A]
In Combobox, I would like to show an action in empty view for users and the user should be able to click

But in the default combobox, when clicked on no items view, the popover is being closed.

Any ideas on how I can make the empty state clickable & disable closing no items view on click

@snowystinger  @devongovett Any ideas in this regard??

**Comment by reidbarber** [comment]
You can use the `renderEmptyState` prop to return a ReactNode containing a link.

**Reply by NarendraKuruva** [reply]
When using `renderEmptyState` prop, and rendering content, though I keep a button in the content, the onclick is not triggering. But the combobox is getting closed

**Reply by snowystinger** [reply]
That makes sense, focus is leaving the input, which closes the popover. There isn't a good way to put a focusable element inside the popover for a combobox. You'd need to build a different component or it won't be accessible. For example, how would a keyboard user access the empty state button?

I suggest a button which triggers a popover. Inside the popover you can put a textfield which controls filtering of a listbox. Then you have keyboard navigation via 'Tab' through all the interactive/focusable elements in the popover.

Another option is to make use of the Links and when your filtered set is empty, add in a link so it's not empty https://react-spectrum.adobe.com/react-aria/ComboBox.html#links which is your action

**Reply by upender7270** [reply]
I am also looking for this kind of feature. With keyboard accessibility compromised, which method is triggering popover close on click renderEmptyState? or I even tried rendering a create element in popover below listbox, but on click still closing popover?

**Reply by upender7270** [reply]
@snowystinger @devongovett   Any ideas in this regard??

**Reply by snowystinger** [reply]
Yes, either of these suggestions https://github.com/adobe/react-spectrum/discussions/6995#discussioncomment-10552976

**Reply by upender7270** [reply]
I think you didn't get my question, What I am trying to understand here is, when I click on renderEmptyState, which method is invoked to set the popover state close?

**Reply by snowystinger** [reply]
First paragraph:
> focus is leaving the input, which closes the popover

This is why you should use one of the methods I outlined. For comboboxes, focus must stay in the input for the popover to be visible.

Let me know if I'm still missing the question.

**Comment by upender7270** [comment]
Ok 
by adding search input and listbox in the popover, how to maintain the search input focused and arrow keys navigation in the listbox without clicking on the tab?

**Reply by snowystinger** [reply]
See if this helps get you going https://github.com/adobe/react-spectrum/discussions/5315
otherwise, searching the Issues/Discussions for "virtual focus" may lead you to more information.
